app/src/main/java/com/simply/birthdayapp/auth/registration/presentation/RegistrationScreen.kt

```diff
@@ -42,6 +56,66 @@ fun RegistrationScreen(viewModel: RegistrationViewModel = koinViewModel()) {
     val passwordError by viewModel.passwordError.collectAsState()
     val repeatedPasswordError by viewModel.repeatedPasswordError.collectAsState()
 
+    val uiState by viewModel.uiState.collectAsState()
+
+    when (uiState) {
+        is SignUpUiState.Success -> {
+
+            Toast.makeText(
+                LocalContext.current,
+                (uiState as SignUpUiState.Success).message,
+                Toast.LENGTH_SHORT
+            ).show()
+            viewModel.resetState()
+            navigateToLanding.invoke()
+        }
+
+        is SignUpUiState.UserExists -> {
+            Toast.makeText(
+                LocalContext.current,
+                (uiState as SignUpUiState.UserExists).message,
+                Toast.LENGTH_SHORT
+            ).show()
+            viewModel.resetState()
+            navigateToLanding.invoke()
+
+        }
+
+        is SignUpUiState.GeneralError -> {
+            val errorMessage = (uiState as SignUpUiState.GeneralError).message
+            AlertDialog(
+                onDismissRequest = {
+                    viewModel.resetState()
+                },
+                title = { Text(text = "Registration Error") },
+                text = { Text(text = errorMessage) },
+                confirmButton = {
+                    TextButton(onClick = {
+                        viewModel.resetState()
+                    }) {
+                        Text(text = "Ok")
+                    }
+                }
+            )
+        }
+
+        is SignUpUiState.Loading -> {
+            Column(
+                modifier = Modifier.fillMaxSize(),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                CircularProgressIndicator(
+                    modifier = Modifier.size(48.dp)
+                )
```

Comment:
Please set color DarkPink for saving app color style

CircularProgressIndicator(
                    modifier = Modifier.size(48.dp),
                    color = DarkPink
                )